sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -209,20 +519,292 @@ node add_malloc_device(void *&data, size_t numBytes, const std::vector<node>& de
+are recorded to a graph object as new nodes for each submission. After recording
+has finished and the queue returns to the executing state, the recorded commands are
+not then executed, they are transparent to any following queue operations.
+
+.Queue State
+image::images/sycl_ext_oneapi_graph/queue-state.svg[]
+
+The state of a queue can be queried with `queue::get_info` using template
+parameter `info::queue::state`. The following entry is added to the
+{queue-info-table}[queue info table] to define this query:
+
+Table 9. Queue info query
+[cols="2a,a,a"]
+|===
+| Queue Descriptors | Return Type | Description
+
+| `info::queue::state`
+| `ext::oneapi::experimental::queue_state`
+| Returns the state of the queue
+
+|===
+
+A default constructed event is returned when a user submits a command-group to
+a queue in the recording state. These events have status
+`info::event_command_status::complete` and a user waiting on them will return
+immediately.
+
+==== Queue Properties
+
+:queue-properties: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:queue-properties
+
+There are {queue-properties}[two properties] defined by the core SYCL
+specification that can be passed to a `sycl::queue` on construction via the
+property list parameter. They interact with this extension in the following
+ways:
+
+1. `property::queue::in_order` - When a queue is created with the in-order
+   property, recording its operations results in a straight-line graph, as each
+   operation has an implicit dependency on the previous operation. However,
+   a graph submitted to an in-order queue will keep its existing structure such
+   that the complete graph executes in-order with respect to the other
+   command-groups submitted to the queue.
+
+2. `property::queue::enable_profiling` - This property has no effect on graph
+   recording. When set on the queue a graph is submitted to however, it allows
+   profiling information to be obtained from the event returned by a graph
+   submission.
+
+For any other queue property that is defined by an extension, it is the
+responsibility of the extension to define the relationship between that queue
+property and this graph extension.
+
+==== New Queue Member Functions
+
+Table 8. Additional member functions of the `sycl::queue` class.
+[cols="2a,a"]
+|===
+|Member function|Description
+
+|
+[source, c++]
+----
+using namespace ext::oneapi::experimental;
+bool queue::begin_recording(command_graph<graph_state::modifiable> &graph)
+----
+
+|Synchronously changes the state of the queue to the `queue_state::recording`
+state.
+
+Parameters:
+
+* `graph` - Graph object to start recording commands to.
+
+Returns: `true` if the queue was previously in the `queue_state::executing`
+state, `false` otherwise.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if the queue is already
+  recording to a different graph.
+
+|
+[source, c++]
+----
+bool queue::end_recording()
```

Comment:
Should we add `using namespace ext::oneapi::experimental;` to all signatures to be consistent?